mantidproject/mantid
# Simple scan explorer

**Description of work.**
This is an interface dedicated to simplifying reducing and analyzing scan data from ILL instruments, especially D16.

This interface is heavily based on the slice viewer widget, reusing the main view. It also modifies tangentially how the reduction operates for this kind of data. 

**To test**
 - In the `Interfaces` menu, go to `ILL/SimpleScanViewer`.  
 - Try opening some scan files (e.g. `066321.nxs`, `066322.nxs` or `generated_data.nxs`) and playing with the options - drawing rectangles, editing, moving or deleting them, both using the mouse and the table.
 - Add a background (e.g. `066320.nxs`) and check the exported peak data has a "I(q) bg corrected" tab, that should correspond to the integration corrected by the background.
 - Extrapolate from one or two rectangles by pressing `f`
 - Export peaks to a TableWorkspace by pressing `p`.
 - Export line plots to workspaces with `x`, `y` and `c`

Also please check that the help button does not interfere with the one in the slice viewer. 

Files for testing can be found there : https://cloud.ill.fr/index.php/s/4BFdANfAE2prXub (it's too big for Github) 

Fixes #34631 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
